File: backend/app/schemas/voice_phishing.py
"""보이스피싱 오디오 분석 API의 응답 계약을 정의한다.

준비된 녹취와 규칙·시퀀스·KoELECTRA 융합 결과를 snake_case 구조로
클라이언트에 전달한다.
"""

from pydantic import BaseModel


class VoicePhishingTurn(BaseModel):
    """분석에 사용한 화자별 발화이다."""

    idx: int
    speaker: str
    text: str


class VoicePhishingAnalysis(BaseModel):
    """오디오 파일에 대응하는 보이스피싱 분석 결과이다."""

    audio_filename: str
    transcript_id: str
    prediction: str
    fusion_score: float
    rule_score: float
    rule_categories: list[str]
    sequence_score: float
    transitions: list[str]
    koelectra_score: float
    decision_reason: str
    turns: list[VoicePhishingTurn]

File: frontend/types/voice_phishing.ts
// 목적: 보이스피싱 판별(voice-phishing) API 응답 타입을 정의한다.
// 주요 역할: Backend Pydantic Schema(VoicePhishingAnalysis)와 동일한 필드명·의미를 유지한다.
//           API JSON 필드는 snake_case 규칙을 유지한다.

export interface VoicePhishingTurn {
  idx: number;
  speaker: string;
  text: string;
}

export interface VoicePhishingAnalysis {
  audio_filename: string;
  transcript_id: string;
  prediction: string;
  fusion_raw_score: number;
  fusion_score: number;
  rule_score: number;
  rule_categories: string[];
  sequence_score: number;
  transitions: string[];
  koelectra_score: number;
  decision_reason: string;
  turns: VoicePhishingTurn[];
}
